classes/class-p4-campaigns.php

```diff
@@ -198,6 +230,11 @@ public function save_taxonomy_meta( $term_id ) {
 			if ( $this->validate( $happypoint_bg_opacity ) ) {
 				update_term_meta( $term_id, $field_id, $happypoint_bg_opacity );
 			}
+
+			$redirect_page = $_POST['redirect_page'] ?? 0;
+			if ( $this->validate_page_types( $redirect_page ) ) {
```

Comment:
Validation is useful here, but this would validate the input against the p4 page types.
You can validate if the passed id corresponds to an actual page (maybe with `get_post()`)